### PR TITLE
cmd/atlascmd: add migrate apply command

### PR DESCRIPTION
**Pull request by masseelch**
This is what it looks like in action for a TTY:
image

And one for text:
image

### CODE REVIEW

**Comment by a8m** [comment]
Please, add screenshots.

File: cmd/atlascmd/migrate.go [hunk 1]
```diff
@@ -134,24 +158,103 @@ func init() {
...
+	if err != nil {
+		return err
+	}
+	// Currently, only in DB revisions are supported.
+	opts := []entmigrate.Option{entmigrate.WithSchema(MigrateFlags.RevisionSchema)}
+	// For SQLite dialect and flavors we have to enable the revision write cache to postpone writing to
+	// the database until the transaction wrapping the migration execution has been committed.
+	if target.Name == "sqlite3" {
+		opts = append(opts, entmigrate.WithCache())
+	}
```
Comment (a8m) [inline cmd/atlascmd/migrate.go:219]: Can't we postpone writing revisions to the end in all dialects? Also, if you still want to keep it here, maybe move it to migrate/migrate.go instead?

**Review by a8m — COMMENTED** [review]
Looks good, before I approve, I added 2 minor questions.

File: cmd/atlascmd/migrate.go [hunk 2]
```diff
@@ -75,6 +83,21 @@ to re-hash the contents and resolve the error
...
 	}
+	// MigrateApplyCmd represents the 'atlas migrate apply' subcommand.
+	MigrateApplyCmd = &cobra.Command{
+		Use:   "apply",
+		Short: "Applies pending migration files on the connected database.",
+		Long: `'atlas migrate apply' reads the migration state of the connected database and computes what migrations are pending.
+It then attempts to apply the pending migration files in the correct order onto the database. 
+The first argument denotes the maximum number of migration files to apply.
+As a safety measure 'atlas migrate apply' will abort with an error, if:
+  - the migration directory is not integer according to the 'atlas.sum' file
```
[the remark this answers is not in the document]
Comment (masseelch) [reply cmd/atlascmd/migrate.go:94]: integer as the adjective to integrity. Maybe this is incorrect english?

[the remark this answers is not in the document]
Comment (a8m) [reply cmd/atlascmd/migrate.go:94]: not in sync*

**Review by a8m — APPROVED** [review]

**Pull request merged**